resources/run_adf.sh

```diff
@@ -139,72 +177,169 @@ if [ $OS = "Linux" ]; then
     export VARIANT=rhel7
     MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_REDHAT
     MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_REDHAT
+    MONGOSH_DOWNLOAD_FILE=$MONGOSH_DOWNLOAD_LINUX_FILE
+    MONGOSH_DOWNLOAD_LINK=$MONGOSH_DOWNLOAD_BASE/$MONGOSH_DOWNLOAD_FILE
   elif [ "$distro" = "\"Ubuntu\"" ]; then
-    export VARIANT=ubuntu1804
+    export VARIANT=ubuntu2204
     MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/linux/$MONGO_DOWNLOAD_UBUNTU
     MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_UBUNTU
+    MONGOSH_DOWNLOAD_FILE=$MONGOSH_DOWNLOAD_LINUX_FILE
+    MONGOSH_DOWNLOAD_LINK=$MONGOSH_DOWNLOAD_BASE/$MONGOSH_DOWNLOAD_FILE
   else
     echo ${distro} not supported
     exit 1
   fi
 elif [ $OS = "Darwin" ]; then
-  MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_MAC
-  MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_MAC
+  if [ $(uname -m) = "x86_64" ]; then
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_MAC
+    MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_MAC
+    MONGOSH_DOWNLOAD_FILE=$MONGOSH_DOWNLOAD_MAC_FILE
+    MONGOSH_DOWNLOAD_LINK=$MONGOSH_DOWNLOAD_BASE/$MONGOSH_DOWNLOAD_FILE
+  else
+    MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/osx/$MONGO_DOWNLOAD_MAC_ARM
+    MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_MAC_ARM
+    MONGOSH_DOWNLOAD_FILE=$MONGOSH_DOWNLOAD_MAC_FILE_ARM
+    MONGOSH_DOWNLOAD_LINK=$MONGOSH_DOWNLOAD_BASE/$MONGOSH_DOWNLOAD_FILE
+  fi
 elif [[ $OS =~ ^CYGWIN ]]; then
   MONGO_DOWNLOAD_LINK=$MONGO_DOWNLOAD_BASE/windows/$MONGO_DOWNLOAD_WIN
   MONGO_DOWNLOAD_FILE=$MONGO_DOWNLOAD_WIN
+  MONGOSH_DOWNLOAD_FILE=$MONGOSH_DOWNLOAD_WINDOWS_FILE
+  MONGOSH_DOWNLOAD_LINK=$MONGOSH_DOWNLOAD_BASE/$MONGOSH_DOWNLOAD_FILE
 else
   echo $(uname) not supported
   exit 1
 fi
 
-check_mongod
-if [[ $? -ne 0 ]]; then
-  if [ $ARG = $START ]; then
+install_mongodb() {
+    (cd $LOCAL_INSTALL_DIR && curl -O $MONGO_DOWNLOAD_LINK)
+    if [[ $OS =~ ^CYGWIN ]]; then
+      unzip -qo $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE -d $LOCAL_INSTALL_DIR 2> /dev/null
+
+      # Obtain unzipped directory name
+      MONGO_UNZIP_DIR=$(unzip -lq $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE | grep mongod.exe | tr -s ' ' \
+              | cut -d ' ' -f 5 | cut -d/ -f1)
+      chmod -R +x $LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR/bin/
+      echo $LOCAL_INSTALL_DIR/$MONGO_UNZIP_DIR
+    else
+      tar zxf $LOCAL_INSTALL_DIR/$MONGO_DOWNLOAD_FILE --directory $LOCAL_INSTALL_DIR
+      # for some reason the arm64 zip contains a directory with aarch64 instead of arm64. Joy.
+      echo $LOCAL_INSTALL_DIR/${MONGO_DOWNLOAD_FILE:0:$((${#MONGO_DOWNLOAD_FILE} - 4))} | sed 's|arm|aarch|'
+    fi
+}
+
+install_mongosh() {
+    (cd $LOCAL_INSTALL_DIR && curl -O $MONGOSH_DOWNLOAD_LINK)
+    if [[ $OS =~ ^CYGWIN ]]; then
+      unzip -qo $LOCAL_INSTALL_DIR/$MONGOSH_DOWNLOAD_FILE -d $LOCAL_INSTALL_DIR 2> /dev/null
+
+      # Obtain unzipped directory name
+      MONGOSH_UNZIP_DIR=$(unzip -lq $LOCAL_INSTALL_DIR/$MONGOSH_DOWNLOAD_FILE | grep mongosh.exe | tr -s ' ' \
```

Comment:
I checked and on the BIC side, I did not need to do anything special
`$LOCAL_INSTALL_DIR/${MONGOSH_DOWNLOAD_FILE:0:$((${#MONGOSH_DOWNLOAD_FILE} - 4))}` should work as far as I can tell.